src/utils/hooks/useGetFullReviewStructure.tsx

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect } from 'react'
+import { FullStructure } from '../types'
+import {
+  ApplicationResponse,
+  Review,
+  ReviewAssignment,
+  ReviewResponse,
+  useGetReviewNewQuery,
+} from '../generated/graphql'
+import addEvaluatedResponsesToStructure from '../helpers/structure/addEvaluatedResponsesToStructure'
+import { useUserState } from '../../contexts/UserState'
+import addOwnedReviewResponse from '../helpers/structure/addOwnedReviewResponse'
+import addElementsById from '../helpers/structure/addElementsById'
+
+interface useGetFullApplicationStructureProps {
+  structure: FullStructure
+  reviewAssignment: ReviewAssignment
+}
+
+const useGetFullReviewStructure = ({
+  structure,
+  reviewAssignment,
+}: useGetFullApplicationStructureProps) => {
+  const [fullStructure, setFullStructure] = useState<FullStructure>()
+  const [isError, setIsError] = useState(false)
+
+  const {
+    userState: { currentUser },
+  } = useUserState()
+
+  const { data, error } = useGetReviewNewQuery({
+    variables: {
+      reviewAssignmentId: reviewAssignment.id,
+      userId: currentUser?.userId || 0,
```

Comment:
> Other hooks are receiving this by props, so it can be checked outside if currentUser is undefined and only receive here an object that is a User type

Hmm, we kind of check that already at the very top route. (so should never get here without userId.

I've cast it to number, that should be better for debugging. Btw I like the idea of using context info in the places where they are needed (vs passing as props):

changed here: 9e6db37b5002f55a4d9a2238e40f9f298d56c885